docs/node/migrations/zero-downtime-migrations/migration-proposal.mdx

```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 1
+sidebar_label: "Migration Proposal"
+id: migration-proposal
+title: Migration Proposal
+description: "Migration Proposal"
+slug: /node/migrations/zero-downtime-migrations/migration-proposal
+---
+
+Migration Declaration is a network governance proposal to coordinate the network migration process. The migration proposal includes the activation period, duration, and the new chain ID that the network will migrate to.
+
+```go
+// MigrationDeclaration creates a new migration. It is used to agree on terms of a migration,
+// and is voted on using Kwil's vote store.
+type MigrationDeclaration struct {
+	// ActivationPeriod is the amount of blocks before the migration is activated.
+	// It starts after the migration is approved via the voting system.
+	// The intention is to allow validators to prepare for the migration.
+	ActivationPeriod uint64
+	// Duration is the amount of blocks the migration will take to complete.
+	Duration uint64
+	// ChainID is the new chain ID that the network will migrate to.
+	// A new chain ID should always be used for a new network, to avoid
+	// cross-network replay attacks.
+	ChainID string
+}
+```
+
+The Migration proposal needs to be approved by atleast 2/3rd of the validators to start the migration process. The migration proposal will get expired if the supermajority of the validators do not approve the proposal within `100800` blocks (roughly 7 days considering the block rate of 6sec). Once the migration proposal is approved, the network moves into migration mode after the specified activation period.
+
+```
```

Comment:
Again, I think the Go code is kind of confusing here. I think you could just have a bullet point for each item on the `MigrationDeclaration` and present it as: "Hey, here are the important components of a migration proposal."

Reply:
I agree this `MigrationResolution`  struct is unnecessary since you've explained it in text above, plus none of it is configurable in any way, unlike the proposal declaration.